service/src/main/java/bio/terra/policy/service/region/RegionService.java

```diff
@@ -69,9 +69,61 @@ public RegionService() {
     constructRegionMapsRecursively(rootRegion);
   }
 
-  public boolean regionContainsDatacenter(String regionName, String datacenterId) {
-    return regionDatacenterMap.containsKey(regionName)
-        && regionDatacenterMap.get(regionName).contains(datacenterId);
+  /**
+   * Get the list of all datacenters available in a region and its subregions, filtered by platform.
+   */
+  @Nullable
+  public HashSet<String> getDataCentersForRegion(String regionName, String platform) {
+    HashSet<String> result = new HashSet<>();
+    HashSet<String> regionDataCenters = regionDatacenterMap.get(regionName);
+
+    if (regionDataCenters != null) {
+      for (String region : regionDataCenters) {
+        if (region.startsWith(platform)) {
+          Datacenter datacenter = datacenterNameMap.get(region);
+          result.add(datacenter.getCode());
+        }
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Gets the ontology starting from the indicated region. Rather than just returning the ontology,
+   * this will filter data centers by the indicated platform.
+   */
+  @Nullable
+  public Region getOntology(String regionName, String platform) {
+    Region mappedRegion = regionNameMap.get(regionName);
+
+    if (mappedRegion == null) {
+      return null;
+    }
+
+    Region result = new Region();
+    result.setName(mappedRegion.getName());
+    result.setDescription(mappedRegion.getDescription());
+
+    List<String> filteredDatacenters = new ArrayList<>();
+    if (mappedRegion.getDatacenters() != null) {
+      for (String datacenter : mappedRegion.getDatacenters()) {
```

Comment:
Might be worth making a method in Region that does this filtering. I feel like I've seen this filter a few places (maybe?)

Reply:
Pulled this out into a function here in the RegionService. Maybe it could go to the region, but right now the Platform only goes to the service level.